docs/sql/statements/insert.rst

```diff
@@ -42,6 +43,23 @@ filled.
 If the expression for any column is not of the correct data type, automatic
 type conversion will be attempted.
 
+The optional RETURNING clause causes INSERT to compute and return value(s)
+based from each row actually inserted (or updated, if an ON CONFLICT DO UPDATE
+clause was used). This is primarily useful for obtaining values that were
+supplied by defaults, such as a :ref:`sequence number
+<sql_administration_system_columns_seq_no>`.
+However, any expression using the table's columns is allowed. The syntax of
+the RETURNING list is identical to that of the output list of SELECT.
+Only rows that were successfully inserted or updated will be returned.
+For example, if a row was locked but not updated because an
+ON CONFLICT DO UPDATE ... WHERE clause condition was not satisfied, the row
```

Comment:
We don't have a `WHERE` clause for `ON CONFLICT... `, do we?

Reply:
good point, thank you.